src/Three.js

```diff
@@ -14,19 +14,38 @@ if ( typeof define === 'function' && define.amd ) {
 
 	module.exports = THREE;
 
+}
+
+
+// this is the variable which should be used to access the global scope
+var scope = undefined;
+
+// the code is running under node environment
+if ( typeof module !== 'undefined' && module.exports ) {
+
+  // if self wasn't defined before, set to nodes global scope
 	if ( typeof self === 'undefined' ) {
 
-		/** @suppress {duplicate, const} */
-		var self = global;
+		scope = global;
 
 	}
+// if the code is running under other environments e.g. webworker, self is defined
+} else if ( typeof self !== 'undefined' ) {
+
+	scope = self;
+
+// the code is running in the browser, so window is defined as the global scope
+} else {
+
+	scope = window;
```

Comment:
If you want to be a bit cleaner, one could create a function on THREE rather than setting a global.  It is less likely to get clubbered by other code.  THREE.getSelf() or something like that.  Just an idea.